bool active
bool leader
---
bool success